### PR TITLE
Feature/update

### PR DESCRIPTION
**Pull request by Buyaki01**
- Add column species to the animals table
- Insert values to the animals table
- Inside a transaction update the animals table by setting the species column to unspecified
- Inside a transaction: Update the animals table by setting the species column to digimon for all animals that have a name ending in mon and Update the animals table by setting the species column to pokemon for all animals that don't have species already set. Commit the transaction.
- Inside a transaction delete all records in the animals table, then roll back the transaction.
- Inside a transaction: Delete all animals born after Jan 1st, 2022. Create a savepoint for the transaction. Update all animals' weight to be their weight multiplied by -1. Rollback to the savepoint. Update all animals' weights that are negative to be their weight multiplied by -1. Commit transaction
- Write queries to the animals table.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -10,4 +10,65 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+
+BEGIN;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+COMMIT;
+
+SELECT name, species FROM animals;
```
Comment (SajjadAhmad14) [inline data.sql]: Please move all these queries written to update the tables to `queries.sql` as everything that is a query that helps to analyze data should be saved in the `queries.sql` file.

File: data.sql [hunk 2]
```diff
@@ -10,4 +10,65 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+
+-- sets all items/rows in the species column to unspecified
+UPDATE animals SET species = 'unspecified';
+
+--verify the changes were made
+SELECT * FROM animals;
+
+-- Undo changes, species column is now empty
+ROLLBACK;
+
```
Comment (SajjadAhmad14) [inline data.sql]: Please write a query to verify that the change was undone after `ROLLBACK`.

File: data.sql [hunk 3]
```diff
@@ -10,4 +10,65 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+BEGIN;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
```
Comment (SajjadAhmad14) [inline data.sql:45]: Please write a query to verify that the change was made before `COMMIT`.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ritta, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql [hunk 4]
```diff
@@ -10,4 +10,71 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+BEGIN TRANSACTION; -- start transaction
+
+UPDATE animals SET species = 'unspecified';
+
+--verify the changes were made
+SELECT * FROM animals;
+
+-- Undo changes
+ROLLBACK;
```
Comment (SajjadAhmad14) [inline data.sql]: Please move all these queries written to update the tables to `queries.sql` as every query that helps to analyze data should be saved in the `queries.sql` file.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ritta,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 4 above)
Comment (Buyaki01) [reply data.sql]: - I have the query SELECT * FROM animals; on line 38 to check whether the column species was updated to 'unspecified' after starting the transaction.

File: data.sql [hunk 5]
```diff
@@ -10,4 +10,68 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+
+SAVEPOINT SPECIES;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SPECIES;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT TRANSACTION;
```
Comment (rukundoeric) [inline data.sql]: - [ ] <b>REQUIRED:</b> As the previous review requested, please move all update-related queries to `queries.sql` as it is requested by the project. As this is your last attempt, I will go ahead and approve the project for the sake of not blocking you for such a small issue, but Kindly do that before you merge.

**Review by rukundoeric — APPROVED** [review]
Hi @Buyaki01,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 5 above)
Comment (Buyaki01) [reply data.sql]: Thank you for your feedback

**Pull request merged**